File: tests/sysid32/reject.m
#import <Foundation/Foundation.h>
#import "MVMavlink.h"
#include <assert.h>

@interface Receiver : NSObject <MVMavlinkDelegate>
@property(nonatomic) unsigned count;
@end
@implementation Receiver
- (void)mavlink:(MVMavlink *)link didGetMessage:(id<MVMessage>)message {
    assert(message.message.msgid == 0 && message.message.sysid == 42);
    self.count++;
}
- (MAV_BOOL)mavlink:(MVMavlink *)link shouldWriteData:(NSData *)data { return YES; }
@end

int main(int argc, char **argv) {
    @autoreleasepool {
        NSString *root = [NSString stringWithUTF8String:argv[1]];
        for (NSString *file in [[NSFileManager defaultManager] contentsOfDirectoryAtPath:root error:nil]) {
            if (![file hasSuffix:@".v2"]) continue;
            MVMavlink *parser = [[MVMavlink alloc] init];
            Receiver *receiver = [[Receiver alloc] init];
            parser.delegate = receiver;
            NSData *data = [NSData dataWithContentsOfFile:[root stringByAppendingPathComponent:file]];
            for (NSUInteger i = 0; i < data.length; i++) {
                [parser parseData:[data subdataWithRange:NSMakeRange(i, 1)]];
            }
            assert(receiver.count == 1);
        }
        for (unsigned flags = 2; flags < 8; flags++) {
            mavlink_message_t msg = {0};
            msg.magic = MAVLINK_STX;
            msg.incompat_flags = flags;
            assert([[MVMessage alloc] initWithCMessage:msg] == nil);
        }
    }
    return 0;
}

File: tests/sysid32/reject.swift
import Foundation
let root = CommandLine.arguments[1]
for file in try FileManager.default.contentsOfDirectory(atPath: root).filter({ $0.hasSuffix(".v1") }) {
    let data = try Data(contentsOf: URL(fileURLWithPath: root + "/" + file))
    let parser = MAVLink()
    var count = 0
    for byte in data {
        if let packet = parser.parse(char: byte, channel: 0) {
            precondition(packet.messageId == 0 && packet.systemId == 42, file)
            count += 1
        }
    }
    precondition(count == 1, file)
}
